File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        
        int get = nums.length/2;
        
        HashMap<Integer, Integer> pokemon = new HashMap<>();
        
        for(int i=0; i<nums.length; i++){
            int cnt = pokemon.getOrDefault(nums[i], 0);
            if(cnt == 0){
                pokemon.put(nums[i], 1);
            }else{
                pokemon.put(nums[i], pokemon.get(nums[i])+1);
            }
        }
        
        int size = pokemon.size();
        int answer = Math.min(size, get);
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    
    var get = Math.floor(nums.length/2);
    var map = new Map();
    
    for(let i=0; i<nums.length; i++){
        if(map.has(nums[i])){
            map.set(nums[i], map.get(nums[i])+1);
        }else{
            map.set(nums[i], 1);
        }
    }
    
    var size = map.size;
    var answer = Math.min(size, get);
    
    return answer;
}